### PR TITLE
Change default server weight

### PR DESCRIPTION
**Pull request by ldez**
### Description

Change default server weight.

Update Oxy:
- fix: Set default weight before calling Weight.

Must be merge after #1782.

Fixes: #1294

### CODE REVIEW

**Comment by emilevauge** [comment]
@containous/traefik this PR is contains potential breaking changes, and we need you help to certify that it is backward compatible. Thanks a lot for your help!

File: types/types_test.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	for _, test := range testCases {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+
+			server := Server{}
+			err := server.UnmarshalTOML(test.data)
+
+			if err != nil && err.Error() != test.expectedErrorMsg {
+				t.Fatalf("UnmarshalTOML(%#v): unexepected error message, got %q, want %s.", test.data, test.expectedErrorMsg, err)
```
Comment (timoreimann) [inline types/types_test.go]: The order of the last two parameters must be swapped.

Comment (timoreimann) [inline types/types_test.go]: Can we improve the message like

`unexpected error or error message: got error %q, want error message %s`

?

Comment (timoreimann) [inline types/types_test.go]: What's the benefit of using `%q` over `%s`?

Comment (ldez) [inline types/types_test.go]: I don't know: it's not my code.

Comment (vdemeester) [inline types/types_test.go]: @timoreimann it escapes stuff doesn't it (and put `"` around)

**Comment by ldez** [comment]
> You also need to change the default weight in the kv template file.
It's already done.

File: types/types_test.go (hunk 1 above)
Comment (timoreimann) [inline types/types_test.go]: @vdemeester Tbh, I can't make full sense out of it even after reading the docs and the example. Not sure how it applies here.

@tanyadegurechaff could you chime in? :)

Comment (timoreimann) [inline types/types_test.go]: Ah, only saw now that @ldez changed it already.

Anyways, please do speak out if you think we should better use `%q`. :-)

File: types/types.go [hunk 2]
```diff
@@ -49,6 +49,34 @@ type Server struct {
...
+		}
+		s.URL = v
+	}
+
+	if w, ok := d["weight"]; ok {
+		v, ok := w.(int64)
+		if !ok {
+			return errors.New("toml: server weight must be int64")
+		}
+		s.Weight = int(v)
```
Comment (tanyadegurechaff) [inline types/types.go]: Because toml allows 64 bit range.

File: types/types_test.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+			name: "toml server with weight 0",
+			data: map[string]interface{}{
+				"weight": int64(0),
+			},
+			expectedServer: Server{
+				Weight: 0,
+			},
+		},
+		{
+			name: "toml server with URL and weight 0",
```
Comment (timoreimann) [inline types/types_test.go]: @ldez WDYT on this one?

**Comment by timoreimann** [comment]
What's the state on this one?

**Comment by ldez** [comment]
@timoreimann I have a problem with the DynamoDB provider.
This supplier has no template, it is a very weird provider.

**Comment by ldez** [comment]
test

**Comment by jgrasl** [comment]
Hi @ldez @timoreimann what's the status on this one? It would be really nice to have this for the backend drain functionality. Can this be in the 1.4 release or it has to go to the next one? Thanks in advance.

**Comment by ldez** [comment]
For now, I close this PR.
I will reopen later.

**Comment by ldez** [comment]
Due to the break changes in Dynamo and Oxy, I move this feature after 1.5.

**Pull request closed without merging**